core/domain/question_services_test.py

```diff
@@ -497,7 +498,7 @@ def _mock_logging_function(msg, *args):
         self.assertEqual(len(observed_log_messages), 1)
         self.assertEqual(
             observed_log_messages[0],
-            'AttributeError \'str\' object has no attribute \'cmd\' %s '
+            'AttributeError \'unicode\' object has no attribute \'cmd\' %s '
```

Comment:
Will this need to be changed back to `str` after the Python 3 migration?

Reply:
Yep

Reply:
Then use `self.assertRegexpMatches(log_msg, 'object has no attribute \'cmd\'')` instead since that's what this test really cares about (not whether it was a str or unicode).

Reply:
Done